pennylane_lightning/core/src/simulators/lightning_gpu/observables/ObservablesGPUMPI.hpp

```diff
@@ -0,0 +1,217 @@
+            new TensorProdObsMPI(std::move(obs))};
+    }
+
+    static auto
+    create(std::vector<std::shared_ptr<Observable<StateVectorT>>> obs)
+        -> std::shared_ptr<TensorProdObsMPI<StateVectorT>> {
+        return std::shared_ptr<TensorProdObsMPI<StateVectorT>>{
+            new TensorProdObsMPI(std::move(obs))};
+    }
+};
+
+/**
+ * @brief General Hamiltonian as a sum of observables.
+ *
+ */
+template <class StateVectorT>
+class HamiltonianMPI final : public HamiltonianBase<StateVectorT> {
+  private:
+    using BaseType = HamiltonianBase<StateVectorT>;
+
+  public:
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    using ComplexT = typename StateVectorT::ComplexT;
+
+    /**
+     * @brief Create a Hamiltonian from coefficients and observables
+     *
+     * @param coeffs Arguments to construct coefficients
+     * @param obs Arguments to construct observables
+     */
+    template <typename T1, typename T2>
+    explicit HamiltonianMPI(T1 &&coeffs, T2 &&obs) : BaseType{coeffs, obs} {}
+
+    /**
+     * @brief Convenient wrapper for the constructor as the constructor does not
+     * convert the std::shared_ptr with a derived class correctly.
+     *
+     * This function is useful as std::make_shared does not handle
+     * brace-enclosed initializer list correctly.
+     *
+     * @param coeffs Argument to construct coefficients
+     * @param obs Argument to construct terms
+     */
+    static auto
+    create(std::initializer_list<PrecisionT> coeffs,
+           std::initializer_list<std::shared_ptr<Observable<StateVectorT>>> obs)
+        -> std::shared_ptr<HamiltonianMPI<StateVectorT>> {
+        return std::shared_ptr<HamiltonianMPI<StateVectorT>>(
+            new HamiltonianMPI<StateVectorT>{std::move(coeffs),
+                                             std::move(obs)});
+    }
+
+    // to work with
+    void applyInPlace(StateVectorT &sv) const override {
+        using CFP_t = typename StateVectorT::CFP_t;
+        DataBuffer<CFP_t, int> buffer(sv.getDataBuffer().getLength(),
+                                      sv.getDataBuffer().getDevTag());
+        buffer.zeroInit();
+
+        for (size_t term_idx = 0; term_idx < this->coeffs_.size(); term_idx++) {
+            DevTag<int> dt_local(sv.getDataBuffer().getDevTag());
+            dt_local.refresh();
+            StateVectorT tmp(dt_local, sv.getNumGlobalQubits(),
+                             sv.getNumLocalQubits(), sv.getData());
+            this->obs_[term_idx]->applyInPlace(tmp);
+            scaleAndAddC_CUDA(
+                std::complex<PrecisionT>{this->coeffs_[term_idx], 0.0},
+                tmp.getData(), buffer.getData(), tmp.getLength(),
+                tmp.getDataBuffer().getDevTag().getDeviceID(),
+                tmp.getDataBuffer().getDevTag().getStreamID(),
+                tmp.getCublasCaller());
+        }
+
+        sv.CopyGpuDataToGpuIn(buffer.getData(), buffer.getLength());
+    }
+};
+
```

Comment:
I'd expect the sparse addition here, and for the single GPU backend, as we have for:
https://github.com/PennyLaneAI/pennylane-lightning-gpu/blob/d4aea775c43d7b7c687dd6448a411bb0fabac10c/pennylane_lightning_gpu/src/algorithms/ObservablesGPU.hpp#L399
and 
https://github.com/PennyLaneAI/pennylane-lightning-gpu/blob/d4aea775c43d7b7c687dd6448a411bb0fabac10c/pennylane_lightning_gpu/src/algorithms/ObservablesGPUMPI.hpp#L415

Reply:
It will be added in PR#526